### PR TITLE
To Do List - Add and edit/remove functionality

### PR DESCRIPTION
**Pull request by hajnaloltyan**
To Do List - Day 2: Add and edit/remove

* Update icons into buttons
* Create new JS file to have functionality: 
               - Add task
               - Edit task's name
               - Remove task
* Saving all to localstorage
* Keep task index updated when other task is removed

* [Live Demo](https://hajnaloltyan.github.io/to-do-list-vjs/)

*Site was deployed by adding scripts to use gh-pages*

### CODE REVIEW

File: src/createTask.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        taskToUpdate.description = taskName.innerText.trim();
+
+        localStorage.setItem('tasks', JSON.stringify(tasks));
+
+        editIcon.classList.remove('fa-floppy-disk');
+        editIcon.classList.add('fa-trash-can');
+      } else {
+        listContainer.removeChild(listItem);
+        const taskIndex = tasks.findIndex((task) => task.id === taskID);
+
```
Comment (FrederickMih) [inline src/createTask.js:80]: - [ ] We are supposed to implement a function for editing task descriptions. Right now, when I try to edit a task, the index of the task replaces the task of the first index when reload as you can see in the short video demo below:


[Screencast from 05-04-2023 11:38:30 PM.webm](https://user-images.githubusercontent.com/63087491/236347614-2467034a-26c9-48cb-9e5a-9cbc58dfbf76.webm)

- Please fix the issue

**Review by FrederickMih — CHANGES_REQUESTED** [review]
## Hi :raising_hand:  @hajnaloltyan, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!





## Highlights :green_circle: 

- PR is professional :+1: 
- Descriptive README.md file :+1: 
- Changes to the Todo List are saved in local storage :+1:  


## Required Changes ♻️
**N/A**
_Check the comments under the review._

## Optional suggestions
**N/A**
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/createTask.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+          };
+          const updatedTasks = [
+            ...tasks.slice(0, taskIndex),
+            updatedTask,
+            ...tasks.slice(taskIndex + 1),
+          ];
+
+          tasks = updatedTasks;
+          localStorage.setItem('tasks', JSON.stringify(tasks));
+        }
```
Comment (NahnahAJ) [inline src/createTask.js:85]: - [ ] Great job fixing the bug the previous reviewer talked about. I see that it's working fine now. However, I noticed that the updated changes aren't getting saved into the local storage, and as a result, when the page is refreshed, the text changes to what it was originally. So kindly fix this as well :100: 
- Video Reference 

[Screencast from 05-05-2023 12:27:45 PM.webm](https://user-images.githubusercontent.com/102699958/236459675-208633e7-e939-4f56-b030-259efae726ac.webm)

**Review by NahnahAJ — CHANGES_REQUESTED** [review]
## Project Status: CHANGES REQUIRED :recycle: 
Hi @hajnaloltyan , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which haven't been addressed yet.


 ## Highlights
- Most requested changes implemented :heavy_check_mark: 
- No linter errors :heavy_check_mark: 
- Professional Pull Request :heavy_check_mark: 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @NahnahAJ in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by medaminedev66** [comment]
Hello @hajnaloltyan 👋,

_This is an additional review as a result of the second opinion request._

I can confirm that the user can edit the task after clicking on the three points icon and then the save icon. The user experience was hard to examined. That's why we encourage you to write a descriptive summary about the changes you worked on so that the code reviewer knows about it.

Anyway, there are some issues with the user experience that need to be addressed. Currently, when a user wants to remove a task, they need to save it first before the remove icon is displayed. However, even after clicking on the remove icon, the task is not deleted from the local storage. We need to fix this issue.

Moreover, users are not able save a new edit of the task after the first edit. This limitation makes the app less user-friendly. To enhance the user experience, we should allow users to edit their tasks at any time. This improvement will make the app more efficient and convenient for users.

Keep rocking 🚀!

**Review by Trast00 — APPROVED** [review]
# Approved 🟢
Hi @hajnaloltyan , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
allfather-approves-approved

## Highlights
- Project look awsome 🍏 🌷 
- Code is clean and well organized 🧼 💯 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

- [ ] **[OPTIONAL]**Kindly, consider adding some element in the **future feature** section of the README.md file. Document your readme file will make your project look much more professional 😊 and help you show a great documentation skill 😄

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@Trast00) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**